### ISSUE TITLE
Integrate Special Topics Form into CAS

### ISSUE THREAD

**Issue by bryantal** [BCStudentSoftwareDevTeam/cas#146]
**[Original report](https://bitbucket.org/laborstudents/cas-flask/issue/146) by Cody Myers (Bitbucket: [cody_myers](https://bitbucket.org/cody_myers), GitHub: [myersCody](https://github.com/myersCody)).**

----------------------------------------

Integrate this [form](https://primary.berea.edu/curriculum-and-advising/curriculum-proposal-course-proposal-forms/) into the Add Course Form inside of CAS.

Attached is a paper copy of the form.

A lot of the inputs in the paper form are already being gathered inside of CAS.

**Comment by bryantal** [comment]
**Original comment by Cody Myers (Bitbucket: [cody_myers](https://bitbucket.org/cody_myers), GitHub: [myersCody](https://github.com/myersCody)).**

----------------------------------------

This feature will require us to add to our current database models in order to record new data related to Special Topics Courses.
The new data:

* Course Description
* Prerequisites
* Proposed Course Cap
* *Major Requirements met with Course*
* *Concentration Requirements met with Course*
* *Minor requirements met with this Course*
* *COGE*

Note that the fields with italics are optional.
I assume that we will be appending these fields to our current course table?

**Comment by bryantal** [comment]
**Original comment by Cody Myers (Bitbucket: [cody_myers](https://bitbucket.org/cody_myers), GitHub: [myersCody](https://github.com/myersCody)).**

----------------------------------------

These fields on our current course form will need to become required when adding a special topics course because they are part of the Special Topics Course Form.

* Course Title
* Course Subject and Number
* Instructor
* Day/Time offered

**Comment by bryantal** [comment]
**Original comment by Cody Myers (Bitbucket: [cody_myers](https://bitbucket.org/cody_myers), GitHub: [myersCody](https://github.com/myersCody)).**

----------------------------------------

The Current Special Topics Form States:

*instructors of such courses should complete this from and forward to the appropriate Program Chair(s) 
who will review submission seek approval from Division Chairs and submit the form to Judy Ginter in the
 Office of the Registrar. Program Chair (with Division Chair approval) or Division Chairs should submit this
 form by email to Judy Ginter. Judy will submit to the Dean for approval, enter the course in Banner, and 
forward the form to the Academic Program Council for their records.*

Due to the wording within the form, I have questions regarding how the flow of the feature should go.

* Should we make it to where the instructors is only allowed to submit special topics forms
* However these courses aren't actually added to the courses listening until they are approved by the division, program, and judy?

**Comment by bryantal** [comment]
**Original comment by Aleksandra Cvetanovska (Bitbucket: [cvetanovskaalex](https://bitbucket.org/cvetanovskaalex), GitHub: [cvetanovskaa](https://github.com/cvetanovskaa)).**

----------------------------------------

Issue #151 was marked as a duplicate of this issue.

**Comment by bryantal** [comment]
**Original comment by Aleksandra Cvetanovska (Bitbucket: [cvetanovskaalex](https://bitbucket.org/cvetanovskaalex), GitHub: [cvetanovskaa](https://github.com/cvetanovskaa)).**

----------------------------------------

We need a new datatable for Special Topic courses. In it, we will store all of the required data for a course, and the data that can be found in the Special Topic form.  Since the Special Topic courses need to be approved before they can be added to the course schedule, we would need to track their status. We can do this by setting an integer: 0 for submitted, 1 for sent to dean, 2 for approved, and 3 for denied.
In order for the admin to monitor which Special Topic course request have been submitted, we need a new interface under "Course Management" where the admin will be able to see all of the Special Topic courses, and to change their information and status. This should look similar to the table found under "Courses". 
Once a Special Topic course has been approved, its data should be duplicated into the Course datatable, and its status set to 2. 
On the Courses page, we should also display the Special Topic courses that are not yet approved, but we need a flag to let the users know what their status is.

**Comment by bryantal** [comment]
**Original comment by Sher Sanginov (Bitbucket: [shersanginov](https://bitbucket.org/shersanginov), GitHub: [sanginovs](https://github.com/sanginovs)).**

----------------------------------------

@cvetanovskaa is this issue resolved? if so, i could not find a pull request for this issue
